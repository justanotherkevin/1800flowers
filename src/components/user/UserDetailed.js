import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class UserDetailed extends Component {
  render() {
    const { user } = this.props;
    return (
      <React.Fragment>
        <div className="user-details col-md-7">
          <h3>{user.description}</h3>
          <p>
            {user.first} {user.last}
          </p>
          <p>Age {user.age}</p>
          <p>Location {user.location}</p>
        </div>
        <div className="user-img col-md-5">
          <img className="img-fluid mx-auto" src="" alt="" />
        </div>
      </React.Fragment>
    );
  }
}

UserDetailed.propTypes = {
  user: PropTypes.object.isRequired
};
