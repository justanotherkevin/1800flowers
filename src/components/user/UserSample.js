import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class UserSample extends Component {
  render() {
    const { user, selectUser } = this.props;
    return (
      <div className="user-container col-lg-4 p-4" key={user.id}>
        <img src="" alt="" className="rounded-circle mb-5" />
        <p>First Name: {user.first}</p>
        <p>Last Name: {user.last}</p>
        <button className="btn btn-primary" onClick={() => selectUser(user)}>
          view details
        </button>
      </div>
    );
  }
}

UserSample.propTypes = {
  user: PropTypes.object.isRequired,
  selectUser: PropTypes.func.isRequired
};
