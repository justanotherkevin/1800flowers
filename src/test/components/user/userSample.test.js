import React from 'react';
import renderer from 'react-test-renderer';
import UserSample from '../../../components/user/UserSample';
import UserDetailed from '../../../components/user/UserDetailed';

const testUser = {
  id: 1,
  first: 'John',
  last: 'Roberts',
  age: 51,
  location: 'Chicago, IL',
  description: 'John is a retired YouTuber'
};
const testFunction = () => {
  console.log('testfunction');
};

it('renders correctly for SampleUser component', () => {
  const tree = renderer
    .create(<UserSample user={testUser} selectUser={testFunction} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders correctly for UserDetailed component', () => {
  const tree = renderer
    .create(<UserDetailed user={testUser}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


