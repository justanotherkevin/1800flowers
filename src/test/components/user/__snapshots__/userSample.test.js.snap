// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for <SampleUser /> component renders correctly 1`] = `
<div
  className="sample-user-container p-4"
>
  <img
    alt=""
    className="rounded-circle sample-image"
    src=""
  />
  <div
    className="sample-info"
  >
    <p>
      First Name: 
      John
    </p>
    <p>
      Last Name: 
      Roberts
    </p>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      view details
    </button>
  </div>
</div>
`;

exports[`for <UserDetailed /> component renders correctly for UserDetailed component 1`] = `
Array [
  <div
    className="user-details col-md-7"
  >
    <h3>
      John is a retired YouTuber
    </h3>
    <p>
      John
       
      Roberts
    </p>
    <p>
      Age 
      51
    </p>
    <p>
      Location 
      Chicago, IL
    </p>
  </div>,
  <div
    className="user-img col-md-5"
  >
    <img
      alt=""
      className="img-fluid mx-auto"
      src=""
    />
  </div>,
]
`;
